src/components/VehicleStatusSelector.js

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import { STATUSES } from "../constants/statuses";
+import { VehicleSelect, VehicleSelectOption } from "./styled";
+
+const VehicleStatusSelector = (props) => {
+  const { stationId, vehicle, handleVehicleStatusChange } = props;
+  const options = [];
+  if (stationId === "4") {
+    options.push(STATUSES.SHIPPED);
+  } else if (stationId === "3") {
+    options.push(STATUSES.COMPLETED);
+  } else {
+    options.push(STATUSES.INPROGRESS, STATUSES.BLOCKED);
+  }
```

Comment:
💡 Options are created based on requirements mentioned in the assignment instruction doc.
1. Status can only be switched between `in processing` or `blocked`
2. Vehicle reaching station 3 should get it's status set to `completed`
3. Once a vehicle reaches the delivery truck, it's status is set to `Delivered`